migration/migration_pkg_test.go

```diff
@@ -160,3 +168,34 @@ func TestChangeTrieNodeEncoding(t *testing.T) {
 		return nil
 	}))
 }
+
+func TestCalculateBlockCommitments(t *testing.T) {
+	testdb := pebble.NewMemTest()
+	t.Cleanup(func() {
+		require.NoError(t, testdb.Close())
+	})
+	chain := blockchain.New(testdb, utils.MAINNET, utils.NewNopZapLogger())
+	client, closeFn := feeder.NewTestClient(utils.MAINNET)
+	t.Cleanup(closeFn)
+	gw := adaptfeeder.New(client)
+
+	for i := uint64(0); i < 3; i++ {
+		b, err := gw.BlockByNumber(context.Background(), i)
+		require.NoError(t, err)
+		su, err := gw.StateUpdate(context.Background(), i)
+		require.NoError(t, err)
+
+		b.EventsBloom = nil
+		require.NoError(t, chain.Store(b, &core.BlockCommitments{}, su, nil))
+	}
+
+	require.NoError(t, testdb.Update(func(txn db.Transaction) error {
+		return calculateBlockCommitments(txn, utils.MAINNET)
+	}))
+
+	for i := uint64(0); i < 3; i++ {
+		b, err := chain.BlockCommitmentsByNumber(i)
+		require.NoError(t, err)
+		assert.False(t, b.TransactionCommitment == nil && b.EventCommitment == nil)
```

Comment:
Shouldn't this be `||` instead of `&&`?

nit: I personally prefer individual `assert.Nil`/`assert.NotNil`s here, but feel free to keep it as it is.

Reply:
Earlier blocks don't have event commitments. So we need to check only TransactionCommitment to be non-nil. Will update